src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -122,26 +124,30 @@ auto get_sshfs_exec_and_options(mp::SSHSession& session)
 // Split a path into existing and to-be-created parts.
 std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
 {
-    QDir complete_path(QString::fromStdString(target));
-    QString absolute;
+    std::string absolute;
 
-    if (complete_path.isRelative())
+    switch (target[0])
     {
-        std::string home = run_cmd(session, "pwd");
-        mp::utils::trim_newline(home);
-        absolute = QString::fromStdString(home) + '/' + complete_path.path();
-    }
-    else
-    {
-        absolute = complete_path.path();
+    case '~':
+        absolute = run_cmd(session, fmt::format("echo {}", target));
```

Comment:
We need to escape here already, no?

Reply:
Absolutely, here and in the `default:` case. And we need to keep the escape later, because the shell answers with unescaped characters.

Another detail is that `target[0]` must not be escaped when it equals `~`, so the shell expands it.